exonum/src/storage/proof_map_index/key.rs

```diff
@@ -511,6 +564,17 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_proof_path_compress_fuzz() {
+        let mut rng = rand::thread_rng();
+        for _ in 0..1000 {
+            let key = random_path(&mut rng);
+            let buf = key.compress();
+            let key2 = ProofPath::decompress(buf.as_ref());
+            assert_eq!(key2, key);
+        }
+    }
```

Comment:
In addition to fuzzing tests I would also recommend to add some test for binary representation. Take some simple proof, manually (on paper) construct the expected binary representation, and then compare the algorithm's output to the result.

It would be nice to cover the edge cases there as well:

* shortest possible proof
* longest possible proof
* proof with bit length that's not a multiple of 8

Reply:
* absent
* `test_proof_path_compress_leaf`
* `test_proof_path_compress_branch`